src/gui/tray/activitylistmodel.cpp

```diff
@@ -576,11 +597,117 @@ void ActivityListModel::triggerAction(int activityIndex, int actionIndex)
+        qCWarning(lcActivity) << "Couldn't find dismiss action in activity at index" << activityIndex
+                              << " actionIndex found " << actionIndex;
+        return;
+    }
+
+    triggerAction(activityIndex, actionIndex);
+}
+
 AccountState *ActivityListModel::accountState() const
 {
     return _accountState;
 }
 
+quint32 ActivityListModel::maxActionButtons()
+{
+    return MaxActionButtons;
+}
+
+QVariantList ActivityListModel::convertLinksToActionButtons(const Activity &activity)
+{
+    QVariantList customList;
+
+    if (activity._links.size() == 1) {
+        return customList;
+    }
+
+    if (static_cast<quint32>(activity._links.size()) > ActivityListModel::maxActionButtons()) {
+        customList << ActivityListModel::convertLinkToActionButton(activity, activity._links.first());
+        return customList;
+    }
+
+    foreach (ActivityLink activityLink, activity._links) {
+        if (activityLink._verb == QStringLiteral("DELETE")
+            || (activity._objectType == QStringLiteral("chat") || activity._objectType == QStringLiteral("call")
+                || activity._objectType == QStringLiteral("room"))) {
+            customList << ActivityListModel::convertLinkToActionButton(activity, activityLink);
+        }
+    }
+
+    return customList;
+}
+
+QVariant ActivityListModel::convertLinkToActionButton(const OCC::Activity &activity, const OCC::ActivityLink &activityLink)
+{
+    auto activityLinkCopy = activityLink;
+
+    const auto isReplyIconApplicable = activityLink._verb == QStringLiteral("WEB")
+        && (activity._objectType == QStringLiteral("chat") || activity._objectType == QStringLiteral("call")
+            || activity._objectType == QStringLiteral("room"));
+
+    const QString replyButtonPath = QStringLiteral("image://svgimage-custom-color/reply.svg");
+
+    if (isReplyIconApplicable) {
+        activityLinkCopy._imageSource =
+            QString(replyButtonPath + "/" + OCC::Theme::instance()->wizardHeaderBackgroundColor().name());
+        activityLinkCopy._imageSourceHovered =
+            QString(replyButtonPath + "/" + OCC::Theme::instance()->wizardHeaderTitleColor().name());
+    }
+
+    const auto isReplyLabelApplicable = activityLink._verb == QStringLiteral("WEB")
+        && (activity._objectType == QStringLiteral("chat")
+        || (activity._objectType != QStringLiteral("room") && activity._objectType != QStringLiteral("call")));
+
+    if (activityLink._verb == QStringLiteral("DELETE")) {
+        activityLinkCopy._label = QObject::tr("Mark as read");
+    } else if (isReplyLabelApplicable) {
+        activityLinkCopy._label = QObject::tr("Reply");
+    }
+
+    return QVariant::fromValue(activityLinkCopy);
+}
+
+QVariantList ActivityListModel::convertLinksToMenuEntries(const Activity &activity)
+{
+    QVariantList customList;
+
+    if (static_cast<quint32>(activity._links.size()) > ActivityListModel::maxActionButtons()) {
```

Comment:
you are already inside the class so no need to use `ActivityListModel::maxActionButtons()` and `maxActionButtons()` is enough